src/Polly.Shared/Retry/RetryPolicyStateWithSleep.cs

```diff
@@ -6,17 +6,23 @@ namespace Polly.Retry
 {
     internal partial class RetryPolicyStateWithSleep : IRetryPolicyState
     {
-        private readonly Action<Exception, TimeSpan, Context> _onRetry;
+        private int _errorCount;
+        private readonly Action<Exception, TimeSpan, int, Context> _onRetry;
         private readonly Context _context;
         private readonly IEnumerator<TimeSpan> _sleepDurationsEnumerator;
 
-        public RetryPolicyStateWithSleep(IEnumerable<TimeSpan> sleepDurations, Action<Exception, TimeSpan, Context> onRetry, Context context)
+        public RetryPolicyStateWithSleep(IEnumerable<TimeSpan> sleepDurations, Action<Exception, TimeSpan, int, Context> onRetry, Context context)
         {
             _onRetry = onRetry;
             _context = context;
             _sleepDurationsEnumerator = sleepDurations.GetEnumerator();
         }
 
+        public RetryPolicyStateWithSleep(IEnumerable<TimeSpan> sleepDurations, Action<Exception, TimeSpan, Context> onRetry, Context context) :
+            this(sleepDurations, (exception, span, i, c) => onRetry(exception, span, context), Context.Empty)
```

Comment:
I think pushing this change is what's causing the tests to fail.  What do you think?